### PR TITLE
Random quote

### PR DESCRIPTION
**Pull request by sebastialimones**

### CODE REVIEW

**Comment by sebastialimones** [comment]
New PR with changes

File: actions/auth_actions.js [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
-  await AsyncStorage.setItem('fb_token', token);
-  dispatch ({ type: FACEBOOK_LOGIN_SUCCESS, payload: token })
 
+export const onAuthStateChange = async (dispatch, getState) => {
+  const handler = (user) => {
+    if (user != null) {
+      dispatch({ type: FACEBOOK_LOGIN_SUCCESS, payload: user.providerData[0] });
+    }
+  };
+  firebase.auth().onAuthStateChanged(handler)
```
Comment (lmuntaner) [inline actions/auth_actions.js]: I did that to make it clearer for you while explaining it.

For actual code, don't create the function and pass it directly to `onAuthStateChanged`.

File: actions/auth_actions.js [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
+    FACEBOOK_ID,
+    { permissions: ['public_profile'] }
+  );
+  if (type === 'cancel'){
+    return dispatch({ type: FACEBOOK_LOGIN_FAIL })
+  } else if (type === 'success') {
+    const credential = firebase.auth.FacebookAuthProvider.credential(token);
+    firebase.auth().signInAndRetrieveDataWithCredential(credential)
+      .catch((error) => {
+        console.log('firebase credentials failed', error)
```
Comment (lmuntaner) [inline actions/auth_actions.js]: You should probably dispatch:
```
dispatch({ type: FACEBOOK_LOGIN_FAIL })
```
here as well

File: src/containers/home.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { facebookLogin } from '../../actions';
+import { connect } from 'react-redux';
+import { View, Text } from 'react-native';
+import RandomQuote from './randomQuote';
+
+
+class Home extends Component{
+
+  componentDidMount (){
+    if (!this.props.isLoggedInLoading && !this.props.user)
```
Comment (lmuntaner) [inline src/containers/home.js]:
```
if (!this.props.isLoggedInLoading && !this.props.user) {
  this.props.facebookLogin()
}
```

File: src/containers/home.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    if (this.props.user) {
+      return (
+        <View>
+          <Text>
+              <RandomQuote />
+          </Text>
+        </View>
+      );
+    }
+    if (this.props.user === 'Failed Log in') {
```
Comment (lmuntaner) [inline src/containers/home.js]: are we still using this? This doesn't need another `if` statement.

It will get this far when `isLoggedInLoading` is `false` and `this.props.user` is falsy.

File: src/containers/home.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+              { 'Loading...' }
+          </Text>
+        </View>
+      );
+    }
+    if (this.props.user) {
+      return (
+        <View>
+          <Text>
+              <RandomQuote />
```
Comment (lmuntaner) [inline src/containers/home.js]: since the `RandomQuote` expect the user and you have it here, I'd say it's better to pass it:
```
<RandomQuote user={ this.props.user } />
```

This means that is not necessary to retrieve in the `mapStateToProps` of `RandomQuote`

File: src/containers/randomQuote.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  componentDidMount(){
+    this.props.fetchRandomQuote(this.props.user)
+  }
+
+  render(){
+    return(
+        <Text>
+          { 
+          !this.props.randomQuote
```
Comment (lmuntaner) [inline src/containers/randomQuote.js]: This should be indented once more. Also go for the truty first, makes it easier to read
```
{
  this.props.randomQuote
    ? this.props.randomQuote.content
    : 'Loading ...'
}
```

File: src/containers/randomQuote.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  }
+}
+
+const mapDispatchToProps = (dispatch) => ({
+  fetchRandomQuote: (user) => dispatch(fetchRandomQuote(user))
+})
+
+const mapStateToProps = (state) => ({ 
+  randomQuote: state.randomQuote, 
+  user: state.auth.user
```
Comment (lmuntaner) [inline src/containers/randomQuote.js]: this won't be needed

File: actions/auth_actions.js [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
+        return dispatch({ type: FACEBOOK_LOGIN_FAIL })
```
Comment (lmuntaner) [inline actions/auth_actions.js]: `return` is not needed. Why did you add it?

Comment (sebastialimones) [reply actions/auth_actions.js]: thought it was a function you have to return from...
